matter_server/server/device_controller.py

```diff
@@ -223,6 +223,13 @@ def thread_credentials_set(self) -> bool:
         """Return if Thread operational dataset as been set."""
         return self._thread_credentials_set
 
+    @property
```

Comment:
it never changes, so you could do `@cached_property`. Or, go one step further and just set `self.loop = …` in constructor.

Reply:
> Or, go one step further and just set self.loop = … in constructor.

That is not possible as the loop on the server is not set at that time.

We do create all objects in constructors, at which time the event loop isn't started yet.

I go for the cached property for now :+1:

Reply:
`@cached_property` is a bit problematic with our API registration function: It used to access the `loop` property before loop as set on server side. I've marked it private (with `_`) and make sure that any private method is skipped (single `_` instead of `__`).